### ISSUE TITLE
remove ql0001_ from the siconos sources

### ISSUE THREAD

**Issue by xhub** [siconos/siconos#251]
Recent developments brought the ConvexQP set of solvers and there exists alternatives like OSQP https://github.com/oxfordcontrol/osqp that are likely to outperform ql0001. Note that ql0001 only soves stricly convex QP.

The plan of action would be:
- change lcp_qp and lcp_nsqp to use a ConvexQP structure and call one of the ConvexQP solver
- remove QP_Solvers.h and the externals/optim_misc/ql0001 directory

Optionally, we could keep the option to use ql0001 via a dynamic load of a libql0001 library. This way it is up to the user to provide such library.

**Comment by vacary** [comment]
ok
I want to mention that the ConvexQP solvers are more or less experimental.

Since ql0001 is based on active sets, I suppose that It would more robust. Let us keep the option for a dynamic load of ql0001

A good option would be also to link with Osqp. Should we distribute is in externals or to use an external installation with a cmake Module ?

**Comment by radarsat1** [comment]
Since solutions seem to depend on quality of QP solvers, maybe it would make sense to be able to select different solvers using iparam?  (and if selected one is missing, simply return error, and iparam=0 should select a safe default)

> A good option would be also to link with Osqp. Should we distribute is in externals or to use an external installation with a cmake Module ?

I guess putting in externals is just for convenience, e.g. i do not see a package for OSQP in debian.  I think the general rule should be to include things for convenience but allow it link to a system version by CMake options, ensuring that this works even if the files are removed from externals.

**Comment by xhub** [comment]
I agree with @radarsat1 : test for a user-provided osqp. If not, compile from source in externals. Let's monitor https://github.com/oxfordcontrol/osqp/issues/88, we may get the cmake file for free.

RIght now, we could use a git submodule in externals to enable a link to OSQP without copying the full source: https://blog.github.com/2016-02-01-working-with-submodules/
